client/src/components/QualityManagement/Parts.jsx

```diff
@@ -0,0 +1,325 @@
+
+  const handlePartNameInput = (e) => {
+    setPartName(e.target.value);
+  };
+
+  const isSelected = (_id) => selected.indexOf(_id) !== -1;
+
+  const handleSelectAllClick = (e) => {
+    if (e.target.checked) {
+      const newSelecteds = data.data.map((n) => n._id);
+      setSelected(newSelecteds);
+      return;
+    }
+    setSelected([]);
+  };
+
+  const handleOnChange = (e, _id) => {
+    const selectedIndex = selected.indexOf(_id);
+    let newSelected = [];
+
+    if (selectedIndex === -1) {
+      newSelected = newSelected.concat(selected, _id);
+    } else if (selectedIndex === 0) {
+      newSelected = newSelected.concat(selected.slice(1));
+    } else if (selectedIndex === selected.length - 1) {
+      newSelected = newSelected.concat(selected.slice(0, -1));
+    } else if (selectedIndex > 0) {
+      newSelected = newSelected.concat(
+        selected.slice(0, selectedIndex),
+        selected.slice(selectedIndex + 1)
+      );
+    }
+
+    setSelected(newSelected);
+  };
+
+  const handleDelete = async (e) => {
+    e.preventDefault();
+    try {
+      await deleteDefects(selected);
+      toast({
+        title: 'Deleted',
+        description: 'Defects have been deleted successfully',
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+      });
+      setSelected([]);
+      refetch();
+    } catch (err) {
+      toast({
+        title: 'Error',
+        description: err.message,
+        status: 'error',
+        duration: 9000,
+        isClosable: true,
+      });
+    }
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setIsLoadingButton(true);
+    try {
+      await createDefect({
+        partName: partName,
+        type: type.current.value,
+        status: status.current.value,
+        description: description.current.value,
+      });
+      toast({
+        title: 'Request Sent',
+        description: 'Defect has been created successfully',
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+      });
+      refetch();
+    } catch (err) {
+      toast({
+        title: 'Error',
+        description: err.message,
+        status: 'error',
+        duration: 9000,
+        isClosable: true,
+      });
+    }
+    setIsLoadingButton(false);
+  };
+
+  if (isLoading) {
+    return <Loader />;
+  }
+
+  const addDefectForm = () => (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>Add a new Defect</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody pb={6}>
+          <FormControl isRequired>
+            <FormLabel>Part Name</FormLabel>
+            <Input onChange={handlePartNameInput} value={partName} placeholder="Part name" />
+          </FormControl>
+
+          <FormControl mt={4}>
+            <FormLabel>Defect Type</FormLabel>
+            <Select ref={type}>
+              <option>Broken</option>
+              <option>Reparable</option>
```

Comment:
I think there is a typo here? Should it be repairable instead of reparable

Reply:
I saw that as well but "reparable" is a legit spelling. So I let it pass